python/BridgingExample/Voice/voice_app.py

```diff
@@ -0,0 +1,205 @@
+        speak_sentence = SpeakSentence(
+            sentence="Please press 1 to accept the call or any other button to send to voicemail",
+            voice="julie"
+        )
+        gather = Gather(
+            gather_url=urljoin(BASE_URL, '/Outbound/Gather'),
+            terminating_digits="#",
+            max_digits=1,
+            first_digit_timeout=10,
+            speak_sentence=speak_sentence,
+            tag=data['tag']
+        )
+        response.add_verb(gather)
+        return response.to_bxml()
+
+
+@voice_routes.route('/Outbound/Gather', methods = ["POST"])
+def handleGather():
+    """
+    Process the result of the gather event and either bridge the calls, or update the A-leg
+    """
+    data = json.loads(request.data)
+    if data['digits'] == '1':
+        response = Response()
+        speak_sentence = SpeakSentence(
+            sentence="The bridge will start now",
+            voice='julie'
+        )
+        bridge = Bridge(data['tag'])
+        response.add_verb(speak_sentence)
+        response.add_verb(bridge)
+        return response.to_bxml()
+    else:
+        updateCall(data['tag'])
+        return ''
+
+
+@voice_routes.route('/Disconnect', methods = ["POST"])
+def handleDisconnect():
+    """
+    Handle any disconnect events related to the B-leg and update the A-leg accordingly
+    """
+    data = json.loads(request.data)
+    if data['eventType'] == 'timeout':
+        updateCall(data['tag'])
+    return ('', 204)
+
+
+@voice_routes.route('/UpdateCall', methods = ["POST"])
+def updateCallRoute():
+    """
+    In the event of a timeout or call screen, update the A-leg call with record bxml to capture a voicemail
+    """
+    data = json.loads(request.data)
+    response = Response()
+    speak_sentence = SpeakSentence(
+        sentence="The person you are trying to reach is not available, please leave a message at the tone",
+        voice='julie'
+    )
+    play_audio = PlayAudio(url="https://www.soundjay.com/button/sounds/beep-01a.wav")
+    record = Record(
+        recording_available_url=urljoin(BASE_URL, '/Recording'),
+        recording_available_method='POST',
+        max_duration=30
+    )
+    response.add_verb(speak_sentence)
+    response.add_verb(play_audio)
+    response.add_verb(record)
+    return response.to_bxml()
+
+
+@voice_routes.route('/Recording', methods = ["POST"])
+def downloadRecording():
+    """
+    Trigger the download of the recorded voicemail upon completion callback from bandwidth
+    """
+    data = json.loads(request.data)
+    with(open('./Recordings/' + str(data['recordingId']) + '.wav', "wb")) as f:
+        response = voice_client.get_stream_recording_media(BANDWIDTH_ACCOUNT_ID, data['callId'], data['recordingId'])
+        f.write(response.body)
+    return ('', 204)
+
+
+@voice_routes.route('/Status', methods = ["POST"])
+def returnStatus():
+    """
+    Capture call status
+    """
+    data = json.loads(request.data)
+    return ('', 204)
```

Comment:
Do you want to do something with `data`?

Reply:
I didnt really think of a case to do anything with the logged status other than save to a DB, i could take the callStatus attribute and log it to screen, or remove data if its unnecessary - there are plenty of examples showing how to get the json so it's not needed here

Reply:
Printing call state to screen